microhapulator/cli/pipe.py

```diff
@@ -0,0 +1,187 @@
+    expected files. Files for samples that pass this test are stored in a list to be passed to
+    Snakemake. (I would prefer to pass the data as a dictionary, but Snakemake complains when the
+    config object is a nested dictionary. So instead we'll reconstruct the dictionary from this
+    list in the Snakefile.)
+    """
+    if suffixes is None:
+        suffixes = (".fastq", ".fastq.gz", ".fq", ".fq.gz")
+    sample_names = check_sample_names(sample_names)
+    files = defaultdict(list)
+    for filepath in traverse(seqpath):
+        if not filepath.name.endswith(suffixes):
+            continue
+        for sample in sample_names:
+            if sample in filepath.name:
+                files[sample].append(filepath)
+    final_file_list = list()
+    for sample in sample_names:
+        filelist = files[sample]
+        if len(filelist) == 2:
+            final_file_list.extend(sorted(filelist))
+        else:
+            message = f"sample {sample}: expected 2 FASTQ files, found {len(filelist)}"
+            raise FileNotFoundError(message)
+    unique_file_names = set([filepath.name for filepath in final_file_list])
+    if len(unique_file_names) != len(final_file_list):
+        raise ValueError("duplicate FASTQ file names found; refusing to proceed")
+    return sample_names, final_file_list
+
+
+def link_or_copy_input(filenames, workdir, docopy=False):
+    seqdir = Path(workdir) / "seq"
+    seqdir.mkdir(parents=True, exist_ok=True)
+    workingfiles = list()
+    for filepath in filenames:
+        workingpath = seqdir / filepath.name
+        workingfiles.append(Path("seq") / workingpath.name)
+        if workingpath.is_file():
+            continue
+        createfunc = copy if docopy else symlink
+        createfunc(filepath.resolve(), workingpath)
+    return [str(wf) for wf in workingfiles]
+
+
+def subparser(subparsers):
+    desc = "Perform a complete end-to-end microhap analysis pipeline"
+    cli = subparsers.add_parser("pipe", description=desc)
+    cli.add_argument(
+        "-w",
+        "--workdir",
+        metavar="D",
+        default=".",
+        help="pipeline working directory; default is current directory",
+    )
+    cli.add_argument(
+        "-n",
+        "--dryrun",
+        action="store_true",
+        help="do not execute the workflow, but display what would have been done",
+    )
+    cli.add_argument(
+        "-t",
+        "--threads",
+        metavar="T",
+        type=int,
+        default=cpu_count(),
+        help="process each batch using T threads; by default, one thread per available core is used",
+    )
+    cli.add_argument(
+        "--copy-input",
+        action="store_true",
+        help="copy input files to working directory; by default, input files are symlinked",
+    )
+    cli.add_argument(
+        "--hg38",
+        default=resource_filename("microhapulator", "data/hg38.fasta.gz"),
+        help=SUPPRESS,
+        # Hidden option for testing purposes
+    )
```

Comment:
Is there a reason this option is still hidden from the user? I accidentally started to download the human reference genome again.

Reply:
I'm trying to make it hard for a naïve user to accidentally mess things up, or really to worry at all how this is managed underneath the hood.

You ran `mhpl8r getrefr` previously, and then it ran again just now?

Reply:
No I hadn't. I hadn't seen the warning about downloading the reference genome until now.

Reply:
So I might be going about this incorrectly but I can't seem to get the `--hg38` option to work. I wanted to use a placeholder file instead of downloading `hg38.fasta.gz`. So I copied `microhapulator/tests/data/refr/usc10-refr.fna` into `microhapulator/data/hg38.fasta.gz` and tried to run the pipeline but it still tried to download the reference genome. Commands I used are:
```
cd microhapulator/
cp tests/data/refr/usc10-refr.fna data/hg38.fasta.gz
cd data
bwa index hg38.fasta.gz
cd ../
mhpl8r pipe -w WDtmp/ -t 16 --hg38 data/hg38.fasta.gz ../microhapulator/tests/data/refr/usc10-refr.fna ../microhapulator/tests/data/def/usc10-offsets.tsv ../microhapulator/tests/data/ gbr-usc10
```

Reply:
I would expect that to work, even without copying it into the One True Location. What error or unexpected behavior are you seeing?

Reply:
(The `mhpl8r getrefr` does a checksum of the `.fasta.gz` file, but Snakemake shouldn't run that command if that and all the index files are present.)

Reply:
Fixed in 8a01609f3b5af846d32cdd7656524cdfff4e1f99.